# fastqc

This is a python script, trying to replicate some and add specific functionality to the FastQC tool[[1]](#1).

## Python version

This script is developed for Python version 3.11.

## Installation

1. Clone repository: `git clone https://github.com/dgruending/fastqc.git`
1. Install requirements: `pip install -r requirements.txt`
1. Call script: `python [-O] fastqc.py -f FILE [args]` Consider using `python -O` for optimization.

## Command line arguments

* [-f/--file] (required): path to an un-/compressed fastq file; multiple files are not supported at this moment
* [-o/--output][default=working directory]: directory for output data (csv files and plots as html files); makes it with parent directories, if they not exist
* [-m/--motifs]: motifs to match, ex. adapter sequences; takes 1+ arguments; arguments need to be separated with at least 1 space
* [-a/--polya]: minimal length of 'A' to be considered a poly A tail; integer > 0
* [-p/--plot][default=False]: toogle plotting
* [-n/--njobs][default=1]: number of processes to spawn for aggregation and evaluation; integer, if less than < 1 the default is used; processes are spawned in addition to the main process
* [-b/--bar]: Sequence coding a barcode at the front of all sequences; Format: `NNXXXNN`, where `N` is any character and is ignored by the program (leading characters are necessary) and `X` marks the Barcode characters (case sensitive)

## Example
The repository contains in the data directory a compressed fastq file with 100,000 entries taken from the first reads of a bigger sequencing run[[2]](#2).

### Example call with the universal illumina adapter as motif
```
python fastqc.py -f data/test.fastq.gz -o test_output/ -a 10 -p -m AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG -b NNNXXXXNN -n 6
```
### Sequence/Barcode quality statistics

The script will produce two tab separated csv files for these statistics with the same layout.
Here the `barcode_quality.csv` generated by the example call.

| Base_position | Mean     | Median | Lower_Quartile | Upper_Quartile | 10th_Percentile | 90th_Percentile | Minimum | Maximum |
|---------------|----------|--------|----------------|----------------|-----------------|-----------------|---------|---------|
| 4             | 36.40137 | 37.0   | 37.0           | 37.0           | 35.0            | 37.0            | 2       | 37      |
| 5             | 36.03226 | 37.0   | 35.0           | 37.0           | 35.0            | 37.0            | 2       | 37      |
| 6             | 34.26815 | 37.0   | 35.0           | 37.0           | 28.0            | 37.0            | 2       | 37      |
| 7             | 35.17154 | 37.0   | 35.0           | 37.0           | 33.0            | 37.0            | 2       | 37      |



# References
<a id="1">[1]</a>
[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
<a id="2">[2]</a>
[SRR24391474](https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&acc=SRR24391474&display=metadata)
